### PR TITLE
Avoid uri encoding problems for component names

### LINKED ISSUE #3175: Sender log in Frank!Console - when there are items, they are not shown when you click the icon
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
The Frank!Console can show sender logs. When a sender log has messages, the number of messages is shown properly next to the icon in the Frank!Console. When you click the icon however, the table with messages is empty.

**Reporter**
Martijn Dirkse

**To Reproduce**
1. Start Frank2Example3 of the Frank!Runner, commit beaf58950306237a81dcb44e994f678e1a1daea1. The F!F version is 7.8-20220512.131203.
2. Stop adapter 04.
3. Go to Test Pipeline. Send some message to adapter 01.
4. Go to the Adapter Status page. Open adapter 01. Look at sender Send. To the right, it has an envelope with the number 1 next to it. This is OK.
5. Click the icon. An empty table appears. This is not OK.

**Screenshots**

image

**Configuration**

The configuration is part of the Frank!Runner. Here are a few lines that illustrate the sender log.

```
			<SenderPipe name="Send">
				<IbisLocalSender javaListener="02 JavaListener"/>
				<JdbcMessageLog slotId="${instance.name}/TestMessageLog"/>
			</SenderPipe>
```

**Input**
```
Please provide an example of the input message. 
Alternatively, provide a Ladybug report.
```

**Environment**
```
FF! 7.8-20220512.131203: Frank2Example3
Running on LAPTOP-R1SK28C using Apache Tomcat/9.0.56
Java Version: OpenJDK Runtime Environment (1.8.0_292-b10)
Heap size: 165M, total JVM memory: 681M
Free disk space: 231GB, total disk space: 464GB
Up since: 2022-05-13 13:31:24 (16m)
```

**Additional Environment**
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql, PostgreSQL]
 - Browser: [e.g. Chrome, Safari, Edge, Firefox]

**Additional Context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

### PR DESCRIPTION
**Pull request by alisihab**
Closes #3175

### CODE REVIEW

File: webapp/src/main/webapp/iaf/gui/js/services.js [hunk 1]
```diff
@@ -887,7 +887,7 @@ angular.module('iaf.beheerconsole')
 			return absolutePath;
 		};
 		this.escapeURL = function(uri) {
-			return encodeURIComponent(uri);
+			return encodeURIComponent(encodeURIComponent(uri));
```
Comment (gvanbrakel) [inline webapp/src/main/webapp/iaf/gui/js/services.js:890]: It should not be necessary to just double encode all uris.

Comment (gvanbrakel) [inline webapp/src/main/webapp/iaf/gui/js/services.js:890]: It should not be necessary to just double encode all uris.

File: core/src/main/java/nl/nn/adapterframework/unmanaged/DefaultIbisManager.java [hunk 2]
```diff
@@ -286,7 +287,7 @@ private void stopAdapters(Configuration configuration) {
 	public Adapter getRegisteredAdapter(String name) {
 		List<Adapter> adapters = getRegisteredAdapters();
 		for (Adapter adapter : adapters) {
-			if (name.equals(adapter.getName())) {
+			if (name.equals(adapter.getName()) || adapter.getName().equals(Misc.urlDecode(name))) {
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/unmanaged/DefaultIbisManager.java:290]: It should not be necessary to explicitly decode adapternames here

Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/unmanaged/DefaultIbisManager.java:290]: It should not be necessary to explicitly decode adapternames here

File: core/src/main/java/nl/nn/adapterframework/webcontrol/api/ShowConfigurationStatus.java [hunk 3]
```diff
@@ -291,7 +291,7 @@ public Response updateAdapter(@PathParam("adapterName") String adapterName, Map<
 				if(value.equals("stop")) { action = IbisAction.STOPADAPTER; }
 				if(value.equals("start")) { action = IbisAction.STARTADAPTER; }
 
-				getIbisManager().handleAction(action, "", adapterName, null, getUserPrincipalName(), false);
+				getIbisManager().handleAction(action, "", adapter.getName(), null, getUserPrincipalName(), false);
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/webcontrol/api/ShowConfigurationStatus.java:294]: Could the name have changed?

Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/webcontrol/api/ShowConfigurationStatus.java:294]: Could the name have changed?

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/core/PipeLine.java [hunk 4]
```diff
@@ -138,7 +138,11 @@ public enum ExitState {
 	}
 
 	public IPipe getPipe(String pipeName) {
-		return pipesByName.get(pipeName);
+		IPipe pipe = pipesByName.get(pipeName);
+		if(pipe == null) {
+			pipe = pipesByName.get(Misc.urlDecode(pipeName));
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/core/PipeLine.java:143]: As discussed, there should be no urlDecode() in PipeLine

**Comment by nielsm5** [comment]
[a-zA-z0-9]

**Comment by jjansenvr** [comment]
We decided not to double escape any names of adapters, pipes and listeners, 

Can you please add the ConfigurationException equivalents to the other names

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**